ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -113,9 +113,9 @@
   ReplicaThread(String threadName, Map<DataNodeId, List<RemoteReplicaInfo>> replicasToReplicateGroupedByNode,
       FindTokenFactory findTokenFactory, ClusterMap clusterMap, AtomicInteger correlationIdGenerator,
       DataNodeId dataNodeId, ConnectionPool connectionPool, ReplicationConfig replicationConfig,
-      ReplicationMetrics replicationMetrics, NotificationSystem notification, StoreKeyFactory storeKeyFactory,
-      StoreKeyConverter storeKeyConverter, Transformer transformer, MetricRegistry metricRegistry,
-      boolean replicatingOverSsl, String datacenterName, ResponseHandler responseHandler, Time time) {
+      ReplicationMetrics replicationMetrics, NotificationSystem notification, StoreKeyConverter storeKeyConverter,
+      Transformer transformer, MetricRegistry metricRegistry, boolean replicatingOverSsl, String datacenterName,
```

Comment:
I also removed `StoreKeyFactory` in PR #1025 .  However, I introduced `StorageManager` into the constructor. This is used for dynamically shutting down store if IO error kept occurring.

Reply:
Will chat with you either about rolling this change back in this PR or merging this with these changes

Reply:
I am ok to keep this change. I could rebase on this change and fix potential conflicts. It shouldn't be difficult.